<launch>

    <!-- Node general configuration -->
    <arg name="camera" default="camera"
        doc="'camera' should uniquely identify the device. All topics are pushed down
                into the 'camera' namespace." />

    <arg name="camera_link_frame_id" default="camera_link"
        doc="Camera link frame identifier." />

    <arg name="color_frame_id" default="camera_color_frame"
        doc="Color camera frame identifier." />

    <arg name="depth_frame_id" default="camera_depth_frame"
        doc="Depth camera frame identifier." />

    <arg name="x" default="0"
        doc="X translation of the depth stream." />
    <arg name="y" default="0"
        doc="Y translation of the depth stream." />
    <arg name="z" default="0"
        doc="Z translation of the depth stream." />

    <!-- Push down all topics/nodelets into "camera" namespace -->
    <group ns="camera">

        <!-- Static transformations -->
        <node pkg="tf2_ros"   
            type="static_transform_publisher"   
            name="camera_depth_tf_publisher"   
            args="$(arg x) $(arg y) $(arg z) 0 0 0 camera_link $(arg depth_frame_id)" />

        <node pkg="tf2_ros"
            type="static_transform_publisher"
            name="camera_color_tf_publisher"
            args="0 0 0 0 0 0 $(arg camera_link_frame_id) $(arg color_frame_id)" />
    </group>
</launch>